zebra/zebra_pbr.c

```diff
@@ -484,7 +484,8 @@ static void zebra_pbr_cleanup_ipset(struct hash_bucket *b, void *data)
 
 	if (ipset->sock == *sock) {
 		hook_call(zebra_pbr_ipset_update, 0, ipset);
-		hash_release(zrouter.ipset_hash, ipset);
+		if (hash_release(zrouter.ipset_hash, ipset))
+			zebra_pbr_ipset_free(ipset);
```

Comment:
isn't bad that `hook_call(zebra_pbr_ipset_update, 0, ipset);` is called inside `zebra_pbr_ipset_free()`?

Reply:
Indeed the hook call is now duplicate, pls fix? :)